conference/src/test/java/org/spine/samples/lobby/conference/ConferenceServiceShould.java

```diff
@@ -212,5 +219,141 @@ public Void apply(Void input) {
+            @Override
+            public Void apply(final PublishConferenceResponse response) {
+                final FindConferenceByIDRequest request = FindConferenceByIDRequest.newBuilder()
+                                                                                   .setId(response.getId())
+                                                                                   .build();
+                conferenceService.findConferenceByID(request, TestStreamObserver.<Conference>newBuilder()
+                                                                                .setNextFunction(checkIsPublished)
+                                                                                .build());
+
+
+                return NO_RESULT;
+            }
+        };
+        final PublishConferenceRequest request = PublishConferenceRequest.newBuilder()
+                                                                         .setId(conference.getId())
+                                                                         .build();
+
+        conferenceService.publish(request, TestStreamObserver.<PublishConferenceResponse>newBuilder()
+                                                             .setNextFunction(publishConferenceHandler)
+                                                             .build());
+
+        final EventStore eventStore = boundedContext.getEventBus()
+                                                    .getEventStore();
+
+        final Function<Event, Void> checkConferencePublished = new Function<Event, Void>() {
+            @SuppressWarnings("ReturnOfNull")
+            @Override
+            public Void apply(Event event) {
+                accumulatedMessages.add(Messages.fromAny(event.getMessage())
+                                                .getClass());
+                return NO_RESULT;
+            }
+        };
+
+
+        final Function<Void, Void> onCompleteUpdate = new Function<Void, Void>() {
+            @Override
+            public Void apply(Void input) {
+                assertTrue(accumulatedMessages.contains(ConferencePublished.class));
+                return NO_RESULT;
+            }
+        };
+
+        final TestStreamObserver conferencePublishedObserver = TestStreamObserver.<Event>newBuilder()
+                                                                                 .setNextFunction(checkConferencePublished)
+                                                                                 .setOnCompleteFunction(onCompleteUpdate)
+                                                                                 .build();
+
+        eventStore.read(EventStreamQuery.newBuilder()
+                                        .build(), conferencePublishedObserver);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void unpublish_conference_and_generate_ConferenceUnpublished_event() {
```

Comment:
The same applies here. As a reader, I want to have an overview on the high level first, and only then (maybe) to dive into details, but you force me to dive first. As you understand, this applies not only to test methods, but for any code.

For example, I don't know anything about cars. First, I want you to give me an overview on the high level: what the car is, why I need it and how can I use it. Instead, you are trying to explain how an engine works on a low-level. Maybe, I'll need it later, but maybe, not. Anyway, I need to know that I can drive a car first.